### PR TITLE
[MM-47825] refactor rolesSystemAdminCmdF function to return an error #21477

### PR DESCRIPTION
**Pull request by yakuter**
#### Summary
Added multi error handing in rolesSystemAdminCmdF function

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-server/issues/21477
JIRA: https://mattermost.atlassian.net/browse/MM-47825

### CODE REVIEW

**Review by noxer — CHANGES_REQUESTED** [review]
Hi @yakuter, thank you very much for your contribution. I'd like to request a few minor changes, other than that it looks very good.

File: commands/roles.go [hunk 1]
```diff
@@ -75,15 +80,17 @@ func rolesSystemAdminCmdF(c client.Client, _ *cobra.Command, args []string) erro
 		if !systemAdmin {
 			roles = append(roles, model.SystemAdminRoleId)
 			if _, err := c.UpdateUserRoles(user.Id, strings.Join(roles, " ")); err != nil {
-				printer.PrintError(fmt.Sprintf("can't update roles for user %q: %s", args[i], err))
+				errStr := fmt.Sprintf("can't update roles for user %q: %s", args[i], err)
+				errs = multierror.Append(errs, errors.New(errStr))
```
[the remark this answers is not in the document]
Comment (yakuter) [reply commands/roles.go]: Solved

**Comment by yakuter** [comment]
@noxer thank you for the review. I solved and commited all the things you mentioned.

File: commands/roles.go [hunk 2]
```diff
@@ -57,10 +58,13 @@ func init() {
 }
 
 func rolesSystemAdminCmdF(c client.Client, _ *cobra.Command, args []string) error {
+	var errs *multierror.Error
 	users := getUsersFromUserArgs(c, args)
 	for i, user := range users {
 		if user == nil {
-			printer.PrintError(fmt.Sprintf("unable to find user %q", args[i]))
+			err := fmt.Errorf("unable to find user %q", args[i])
```
Comment (noxer) [inline commands/roles.go:65]: You can use `%w` to wrap errors in `fmt.Errorf`.

File: commands/roles.go [hunk 3]
```diff
@@ -75,15 +79,17 @@ func rolesSystemAdminCmdF(c client.Client, _ *cobra.Command, args []string) erro
 		if !systemAdmin {
 			roles = append(roles, model.SystemAdminRoleId)
 			if _, err := c.UpdateUserRoles(user.Id, strings.Join(roles, " ")); err != nil {
-				printer.PrintError(fmt.Sprintf("can't update roles for user %q: %s", args[i], err))
+				err := fmt.Errorf("can't update roles for user %q: %s", args[i], err)
```
Comment (noxer) [inline commands/roles.go:82]: Same here

**Review by noxer — CHANGES_REQUESTED** [review]
Just two minor adjustments, then we're done ;)

File: commands/roles.go (hunk 3 above)
Comment (noxer) [reply commands/roles.go]: And you should rename the error instead of shadowing `err`.

Comment (yakuter) [reply commands/roles.go]: Ohh sorry I missed that. Did it now. Thank you.

**Review by noxer — APPROVED** [review]
Perfect, thank you very much!

**Comment by yakuter** [comment]
> Perfect, thank you very much!

Thanks to you @noxer :)

**Review by isacikgoz — APPROVED** [review]
Awesome, thanks for the contribution @yakuter 🎉

**Comment by yakuter** [comment]
Thanks for your support for my first PR to mattermost @noxer @isacikgoz 🥳

**Pull request merged**